app/controllers/dogs_controller.rb

```diff
@@ -0,0 +1,57 @@
+class DogsController < ApplicationController
+  def new
+    @user = User.find(params[:id])
+    @dog = Dog.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @dog = Dog.new(dog_params)
+    @dog.user = current_user
+    if @dog.save
+      flash[:success] = 'New Dog Added!'
+      redirect_to profile_path(current_user)
+    else
+      flash[:errors] = @dog.errors.full_messages.join(". ")
+      redirect_to profile_path(current_user)
+    end
+  end
+
+  def edit
+    dog = Dog.find(params[:id])
+    if signed_in? && current_user == dog.user
+      @dog = Dog.find(params[:id])
+    elsif !signed_in?
+      authenticate_user!
+    else
+      flash[:alert] = 'You have no permission to edit this dog.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.